### PR TITLE
Feature/fe answering process

### PR DESCRIPTION
**Pull request by sergiy17**
[Link to JIRA ticket](https://ssu-jira.softserveinc.com/browse/LR-)

### Author.

- [ ] I have added link to this PR to the JIRA
- [ ] I have logged time in JIRA.

### Reviewers.

- [ ] @reviewer1
- [ ] @reviewer2
- [ ] @vovka

### Issue summary.

*Short description of you feature or bug.*

### Fix summary. 

_Short description of how you implemented a feature or fixed a bug. And why._

### Testing approach.

_Please describe your testing approach here._
* _What cases do your tests cover?_
* _Did you cover all of the edge cases?_
* _What kinds of tests did you write?_
  * _unit tests?_
  * _controller tests?_
  * _integration tests?_
* _What was your rationale for choosing one kind of test over another for testing each case?_

### CODE REVIEW

File: spec/routing/answers_routing_spec.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+RSpec.describe AnswersController, type: :routing do
+  describe "routing" do
+
+    it "routes to #new" do
+      expect(:get => "/answers/new").to route_to("answers#new")
+    end
+
+    it "routes to #create" do
+      expect(:post => "/answers").to route_to("answers#create")
```
Comment (vovka) [inline spec/routing/answers_routing_spec.rb]: Codacy Issue found: [Use the new Ruby 1.9 hash syntax.](https://www.codacy.com/app/vovka/video_quiz/file/2606429221/issues/source?bid=3205144&fileBranchId=3309687#l11)

File: spec/routing/answers_routing_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+require "rails_helper"
+
+RSpec.describe AnswersController, type: :routing do
+  describe "routing" do
+
```
Comment (vovka) [inline spec/routing/answers_routing_spec.rb]: Codacy Issue found: [Extra empty line detected at block body beginning.](https://www.codacy.com/app/vovka/video_quiz/file/2606429221/issues/source?bid=3205144&fileBranchId=3309687#l5)

File: spec/routing/answers_routing_spec.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    it "routes to #new" do
+      expect(:get => "/answers/new").to route_to("answers#new")
```
Comment (vovka) [inline spec/routing/answers_routing_spec.rb]: Codacy Issue found: [Use the new Ruby 1.9 hash syntax.](https://www.codacy.com/app/vovka/video_quiz/file/2606429221/issues/source?bid=3205144&fileBranchId=3309687#l7)

File: spec/controllers/answers_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+# sticking to rails and rspec-rails APIs to keep things simple and stable.
+#
+# Compared to earlier versions of this generator, there is very limited use of
+# stubs and message expectations in this spec.  Stubs are only used when there
+# is no simpler way to get a handle on the object needed for the example.
+# Message expectations are only used when there is no simpler way to specify
+# that an instance is receiving a specific message.
+
+RSpec.describe AnswersController, type: :controller do
+
```
Comment (vovka) [inline spec/controllers/answers_controller_spec.rb]: Codacy Issue found: [Extra empty line detected at block body end.](https://www.codacy.com/app/vovka/video_quiz/file/2606429207/issues/source?bid=3205144&fileBranchId=3309687#l22)

File: spec/controllers/answers_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+      get :new, {:quiz_id => @quiz.id}, valid_session
+      expect(assigns(:answer)).to be_a_new(Answer)
+    end
+  end
+
+  describe "POST #create" do
+    context "with valid params" do
+      it "creates a new Answer" do
+        expect {
+          post :create, {:quiz_id => @quiz.id, :answer => valid_attributes}, valid_session
```
Comment (vovka) [inline spec/controllers/answers_controller_spec.rb]: Codacy Issue found: [Use the new Ruby 1.9 hash syntax.](https://www.codacy.com/app/vovka/video_quiz/file/2606429207/issues/source?bid=3205144&fileBranchId=3309687#l59)

File: spec/controllers/answers_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+      it "redirects to the quiz_interviews_show_path" do
+        post :create, {:quiz_id => @quiz.id, :answer => valid_attributes}, valid_session
+        get :new, {:quiz_id => @quiz.id}, valid_session
+        expect(response).to redirect_to(quiz_interviews_show_path)
+      end
+    end
+
+    context "with invalid params" do
+      it "assigns a newly created but unsaved answer as @answer" do
+        post :create, {:quiz_id => @quiz.id, :answer => invalid_attributes}, valid_session
```
Comment (vovka) [inline spec/controllers/answers_controller_spec.rb]: Codacy Issue found: [Line is too long. [90/80]](https://www.codacy.com/app/vovka/video_quiz/file/2606429207/issues/source?bid=3205144&fileBranchId=3309687#l78)

Comment (vovka) [inline spec/controllers/answers_controller_spec.rb]: Codacy Issue found: [Use the new Ruby 1.9 hash syntax.](https://www.codacy.com/app/vovka/video_quiz/file/2606429207/issues/source?bid=3205144&fileBranchId=3309687#l78)

File: spec/controllers/answers_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+  # This should return the minimal set of attributes required to create a valid
+  # Answer. As you add validations to Answer, be sure to
+  # adjust the attributes here as well.
+  before(:each) do
+    @user = User.create!(email: "user@example.org", name: "Name")
+    sign_in @user
+    @quiz = Quiz.create!(name: 'This is quiz #1')
+    @interview = Interview.create!(quiz_id: 1, user_id: 1, current_question_id: 1)
+    @quiz.questions.create!(content: 'This is question #1',record_time_limit: 10)
```
Comment (vovka) [inline spec/controllers/answers_controller_spec.rb]: Codacy Issue found: [Line is too long. [82/80]](https://www.codacy.com/app/vovka/video_quiz/file/2606429207/issues/source?bid=3205144&fileBranchId=3309687#l31)

File: spec/controllers/answers_controller_spec.rb (hunk 5 above)
Comment (vovka) [inline spec/controllers/answers_controller_spec.rb]: Codacy Issue found: [Line is too long. [90/80]](https://www.codacy.com/app/vovka/video_quiz/file/2606429207/issues/source?bid=3205144&fileBranchId=3309687#l59)

File: spec/controllers/answers_controller_spec.rb [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+RSpec.describe AnswersController, type: :controller do
+
+  # This should return the minimal set of attributes required to create a valid
+  # Answer. As you add validations to Answer, be sure to
+  # adjust the attributes here as well.
+  before(:each) do
+    @user = User.create!(email: "user@example.org", name: "Name")
+    sign_in @user
+    @quiz = Quiz.create!(name: 'This is quiz #1')
+    @interview = Interview.create!(quiz_id: 1, user_id: 1, current_question_id: 1)
```
Comment (vovka) [inline spec/controllers/answers_controller_spec.rb]: Codacy Issue found: [Line is too long. [82/80]](https://www.codacy.com/app/vovka/video_quiz/file/2606429207/issues/source?bid=3205144&fileBranchId=3309687#l30)
